doc/developer/design/20220819_bounded_input_reliance.md

```diff
@@ -0,0 +1,193 @@
+safe to restart a specific source at.
+- _upstream commit frontier_: A _source gauge_ (whose type is `FromTime`, e.g. a Postgres LSN or a Kafka `(partition, offset)`) frontier
+that can be safely committed to the upstream service. The details and exact semantics of this commit frontier
+are specific to the source type, and some sources may ignore any such frontier.
+- _reclock collection_: A pTVC that stores the (bi-directional) mapping between a _source_gauge_ and _target_gauge_.
+- source: (or "upstream source" or "upstream service") a service like Kafka or Postgres that produces a stream of data.
+- `SourceReader`: as in "a `SourceReader` implmentator", the Rust type that produces actual data from an upstream source.
+- `Source Ingestion operator`: The Timely operator that drives a `SourceReader`
+- `RemapOperator`: The Timely operator that produces the _relock collection_
+- `ReclockOperator`: The Timely operator that consumes the _reclock collection_ to produce timestamped data for a source.
+
+[the original reclocking design doc]: https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/20210714_reclocking.md
+
+## Goals
+
+- Primary Goal:
+  - Regularly commit advancing _upstream commit frontiers_ for all sources
+- Additional Goals:
+  - Restart sources with up-to-date _resumption frontiers_.
+    - Note that this is _required_, as the upstream sources could compact data past the _upstream commit frontier_.
+  - Compact the _reclock collection_
+  - Localize _resumption frontier_ and _upstream commit frontier_ management inside `storaged` instances
+  (i.e. not in the storage controller).
+
+## Non-goals
+
+- Expose the _reclock collection_ to sql (for now).
+
+## Overview
+
+To accomplish the **Primary Goal** for a source, Materialize will semi-regularly providing an _upstream commit frontier_
+to the underlying `SourceReader`, which will communicate that frontier to the upstream source. After such a frontier
+has been exchanged, the upstream source may choose to forgo providing data from before that frontier when Materialize
+restarts ingestion for that source.
+
+**Source types that do not provide such an API may not be able to provided _bounded input reliance_**
+
+The _upstream commit frontier_ can be derived from the source's _resumption frontier_, by mapping the
+_resumption frontier_ to the _upstream commit frontier_ using the sources _reclock collection_.
+
+The _resumption frontier_ can be calculating by combining the `uppers` of all persist shards that are related to this source
+(see the [persist design doc] for more information about these `uppers`). Currently this is only:
+- The "data" persist shard.
+- The _reclock collection_ persist shard.
+
+In the future this will contain other shards like "intermediate state shards".
+The specifics of the "combination" is unspecified in this
+document, but is most likely equivalent to a [`Lattice::meet`].
+
+## Additional Goals
```

Comment:
nit: Maybe make this `###` (and indent it's children further), to keep it under _Overview_.